### PR TITLE
glue with the dev backend works again

### PR DESCRIPTION
**Pull request by folkertdev**
couple of small extra fixes were needed to make this work again

### CODE REVIEW

File: crates/compiler/gen_dev/src/object_builder.rs [hunk 1]
```diff
@@ -29,6 +29,28 @@ pub fn build_module<'a, 'r>(
...
+
+    if std::env::var("ROC_DEV_WRITE_OBJ").is_ok() {
+        let module_out = module_object
+            .write()
+            .expect("failed to build output object");
+
+        let file_path = std::env::temp_dir().join("app.o");
+        println!("gen-test object file written to {}", file_path.display());
+        std::fs::write(&file_path, module_out).expect("failed to write object to file");
+    }
```
Comment (folkertdev) [inline crates/compiler/gen_dev/src/object_builder.rs:43]: moved here so it is evaluated whenever we use the dev backend (dev tests, repl, or glue with dev)

File: crates/compiler/gen_dev/src/generic64/mod.rs [hunk 2]
```diff
@@ -2949,7 +2949,11 @@ impl<
         self.load_layout_alignment(list_layout, Symbol::DEV_TMP);
 
         // Load element_width argument (usize).
-        self.load_layout_stack_size(*ret_layout, Symbol::DEV_TMP2);
+        let element_layout = match self.interner().get_repr(*ret_layout) {
+            LayoutRepr::Builtin(Builtin::List(e)) => e,
+            _ => unreachable!(),
+        };
+        self.load_layout_stack_size(element_layout, Symbol::DEV_TMP2);
```
Comment (folkertdev) [inline crates/compiler/gen_dev/src/generic64/mod.rs:2956]: `List.append` was using the wrong layout here, causing all sorts of memory issues

File: crates/glue/src/load.rs [hunk 3]
```diff
@@ -130,46 +130,19 @@ pub fn generate(
...
-                            Ok(x) => x.into(),
-                        };
-
-                    let files = files.unwrap_or_else(|err| {
-                        eprintln!("Glue generation failed: {err}");
-
-                        process::exit(1);
-                    });
+
+                    // NOTE: DO NOT DROP LIB! the return value will include static roc strings that
```
Comment (bhansconnect) [inline crates/glue/src/load.rs:136]: Haha. wow. That's a good catch.

File: crates/glue/src/load.rs [hunk 4]
```diff
@@ -242,6 +215,69 @@ pub fn generate(
...
+) -> roc_std::RocList<roc_type::File> {
+    let roc_call_result = match backend {
+        CodeGenBackend::Assembly(_) => {
+            type MakeGlueReturnType = RocCallResult<
+                roc_std::RocResult<roc_std::RocList<roc_type::File>, roc_std::RocStr>,
+            >;
+            type MakeGlue =
+                unsafe extern "C" fn(roc_std::RocList<roc_type::Types>) -> MakeGlueReturnType;
+
+            let name_of_main = "test_main";
```
Comment (bhansconnect) [inline crates/glue/src/load.rs:231]: So we don't have a proper main/_exposed yet, but this is just a workaround?

**Review by bhansconnect — APPROVED** [review]

File: crates/glue/src/load.rs (hunk 3 above)
Comment (folkertdev) [reply crates/glue/src/load.rs:136]: yeah, I was not careful and that will just segfault. even a later explicit drop of the lib causes issues, so something to keep in mind.

File: crates/glue/src/load.rs (hunk 4 above)
Comment (folkertdev) [reply crates/glue/src/load.rs:231]: we do, this is proper in that it uses the same mechanism as the gen tests. We need the setjmp/longjmp insertion here. 

but yeah this needs to be made more consistent at some point

**Pull request merged**